internal/tofu/context_validate_test.go

```diff
@@ -2487,3 +2488,84 @@ locals {
 		t.Fatalf("expected deprecated warning, got: %q\n", warn)
 	}
 }
+
+func TextContext2Validate_providerFunctions(t *testing.T) {
```

Comment:
`TestThingBeingTested_whatAboutItIsBeingTested` is a naming scheme used all over this codebase, so unless we want to go rewrite that everywhere I'd suggest finding a way to disable this rule through the whole codebase, rather than just in this one spot.

I don't actually know where this naming scheme came from -- this codebase has been using this style since before I was working on it -- but the `testing` package documentation describes a vaguely-similar naming scheme for "examples" in test files and so I'd guess the early authors of this codebase chose to kinda-generalize that pattern to the `Test...` prefix too.

(I'd agree that the specific way it's used for the _context_ tests is awkward -- literally hundreds of tests all have the same prefix before the underscore. But I also don't think it's worth the churn and history noise to "correct" them all, nor worth the nolint-comment-noise to disable the rule for each new test matching the existing scheme.)